### PR TITLE
Low hanging fruit

### PR DESCRIPTION
**Pull request by EricWVGG**
- removed unused Perks component
- FeaturedArtPanel correctly formats starting/ending grant dates
- AccountButton is no longer clickable before JWT is read, and presents a spinner
- UploadFileControl improvements…
  - uses InviseFileInput from AvatarUpload to for drag-and-drop images and DRY
  - uses existing text inputs to show cloudinary url
  - shows image and video previews on right

old UploadFileControl had some notes about memory leaks, but didn't actually do anything about them. Let me know if I should deeper investigate.

### CODE REVIEW

**Review by rubelux — CHANGES_REQUESTED** [review]
In `page/admin/grants` line 50, it gets stuck on the spinning component, it should only display the spinner if there is  user session being loaded otherwise it needs to display a message inviting user to login "with a curator account"

**Review by rubelux — CHANGES_REQUESTED** [review]
in `pages/admin/grants/` fix the rollover call on dark theme

Screenshot 2022-12-21 at 11 31 16

**Review by rubelux — CHANGES_REQUESTED** [review]
in `pages/admin/grants/` Set a maximum height to the list container and make the item scroll vertically within

File: src/components/Form/Controls/UploadFileControl/index.tsx [hunk 1]
```diff
@@ -1,121 +1,95 @@
...
-            style={{ display: 'none' }}
-            id={path}
-            className={className}
-          />
-          <label htmlFor={path}>Upload</label>
-        </>
-      )}
-
-      {isThereImg && (
-        <ImageWrapper>
```
Comment (rubelux) [inline src/components/Form/Controls/UploadFileControl/index.tsx:75]: * The image wrapper is not expanding to 300px 
* Make the component height to 300px 
* hide the image link 
* Hide the component label
* Center the uploaded image in the centre of the box



Screenshot 2022-12-21 at 11 58 41

**Review by rubelux — CHANGES_REQUESTED** [review]

File: src/components/Form/Controls/UploadFileControl/index.tsx (hunk 1 above)
Comment (EricWVGG) [reply src/components/Form/Controls/UploadFileControl/index.tsx:75]: The design of this was deliberate, I wanted to keep it consistent with our existing forms and have the input accurately reflect what was being recorded. Also if we want a proper design, I think that would be a task for Simon or Juju, and I don't want to put anything on their plate right now.

Would you mind if I instead made the image preview clickable, and it launches a popup where the asset is full-sized? We already have the component for that, it should be an easy drop-in.

Comment (EricWVGG) [reply src/components/Form/Controls/UploadFileControl/index.tsx:75]: I've added a preview modal for media. I think you'll like it!

File: src/components/Form/Controls/UploadFileControl/index.tsx [hunk 2]
```diff
@@ -1,121 +1,95 @@
...
+        type="string"
+        placeholder={uischema?.options?.placeholder || ' '}
+        defaultValue={data}
+        value={data}
+      />
+      <PreviewWrapper visible={!!selectedFile && !!data}>
+        {fileType(selectedFile) === 'image' && !!data && <PreviewImage src={data} />}
+        {fileType(selectedFile) === 'video' && !!data && (
+          <video loop={true} autoPlay={true} controls={false} muted={true}>
+            <source src={data} type="video/mp4" />
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/Form/Controls/UploadFileControl/index.tsx:79]: @EricWVGG I do not see WebM in the Video type list

File: src/components/Form/Controls/UploadFileControl/index.tsx [hunk 3]
```diff
@@ -1,121 +1,95 @@
...
+}: Partial<StringControlProps> & UploadFileControlProps) => {
   const [selectedFile, setSelectedFile] = useState<File>()
-
-  // create a preview as a side effect, whenever selected file is changed
-  // preventing memory leaks with createObjectURL
   useEffect(() => {
-    if (!selectedFile) {
-      setTempImg('')
+    if (!selectedFile) return
+    uploadFile(selectedFile)
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/Form/Controls/UploadFileControl/index.tsx:30]: @EricWVGG this is still happening

File: src/components/Form/Controls/UploadFileControl/index.tsx (hunk 1 above)
Comment (rubelux) [reply src/components/Form/Controls/UploadFileControl/index.tsx:75]: All the NFTs assets are all square, so we need these boxes to be square now. Can you make them square? 
Also, they label and the text is still shows up,

**Review by rubelux — CHANGES_REQUESTED** [review]
@EricWVGG VIdeos are to uploaded to Cloudinary, Component shows an error

File: src/components/Form/Controls/UploadFileControl/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/Form/Controls/UploadFileControl/index.tsx:79]: Cloudinary will deliver WebM to Chrome browsers

File: src/components/Form/Controls/UploadFileControl/index.tsx (hunk 1 above)
Comment (EricWVGG) [reply src/components/Form/Controls/UploadFileControl/index.tsx:75]: ^ try clicking the image for a preview popup. Much better than 300px wrapper.

The preview boxes in the corner do not force square images.

The URL is there in the value for a reason — it's what we're actually storing.

**Comment by EricWVGG** [comment]
> VIdeos are to uploaded to Cloudinary, Component shows an error

please clarify with steps

**Pull request merged**